Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -354,7 +356,7 @@ public IEnumerable<string> ListFiles(string path, SearchOption searchOption, par
 
             using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
             {
-                var files = repo.Diff.Compare<TreeChanges>(null, DiffTargets.Index, lookupList, compareOptions: new CompareOptions() { IncludeUnmodified = true, Similarity = SimilarityOptions.None })
+                var files = repo.Diff.Compare<TreeChanges>(null, DiffTargets.Index, lookupList, new ExplicitPathsOptions { ShouldFailOnUnmatchedPath = false }, compareOptions: new CompareOptions() { IncludeUnmodified = true, Similarity = SimilarityOptions.None })
```

Comment:
It's for searching for any files in the repo for type discovery. We use git ls-files instead of a normal `Directory.GetFiles()` because of performance impact when we have a massive node_modules folder.